app/user_cleanup.go

```diff
@@ -0,0 +1,135 @@
+package app
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/systemaccount"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type userCleanup struct {
+	users         v3.UserInterface
+	userLister    v3.UserLister
+	clusterLister v3.ClusterLister
+	projectLister v3.ProjectLister
+}
+
+func cleanupOrphanedSystemUsers(management *config.ManagementContext) {
+	u := userCleanup{
+		users:         management.Management.Users(""),
+		userLister:    management.Management.Users("").Controller().Lister(),
+		clusterLister: management.Management.Clusters("").Controller().Lister(),
+		projectLister: management.Management.Projects("").Controller().Lister(),
+	}
+
+	go wait.PollImmediate(time.Hour*24, 0, func() (bool, error) {
+		steps := 5
+		backOffDuration := time.Minute * 10
+		factor := 2
+		var err error
+		for steps > 0 {
+			err = u.cleanup()
+			if err != nil {
+				time.Sleep(backOffDuration)
+				backOffDuration = time.Duration(factor) * backOffDuration
+			} else {
+				break
+			}
+			steps--
+		}
+		if err != nil {
+			// returning false, nil because PollImmediate terminates on error
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
```

Comment:
Can we add some debug log statements when PollImmediate is invoked and when a systemuser is deleted by the poll. This will help in testing and validating cleanup.